zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/KeyringCache.java

```diff
@@ -38,9 +38,9 @@ public void put(User user, Session session) throws IOException {
                 keyringMap.put(session, user.keyring());
 
                 // populate the scheduler keyring
-                for (String collectionId : user.keyring.list()) {
+                for (String collectionId : user.keyring().list()) {
                     if (!schedulerCache.containsKey(collectionId))
-                        schedulerCache.put(collectionId, user.keyring.get(collectionId));
+                        schedulerCache.put(collectionId, user.keyring().get(collectionId));
```

Comment:
This could be written as

```
user.keyring().list().stream()
    .filter(c -> !schedulerCache.containsKey(c))
    .forEach(c -> schedulerCache.put(c, user.keyring().get(c)));
```